test/integration/infra.go

```diff
@@ -112,17 +112,32 @@ func (infra *infra) setup() error {
 		return err
 	}
 
-	if infra.UseInitializer {
-		infra.InjectionPolicy = inject.InjectionPolicyOptOut
+	mesh, err := inject.GetMeshConfig(client, infra.IstioNamespace, "istio")
+	if err != nil {
+		return err
+	}
 
-		// NOTE: InitializerConfiguration is cluster-scoped and may be
-		// created and used by other tests in the same test
-		// cluster.
+	infra.InjectConfig = &inject.Config{
+		Policy:    inject.InjectionPolicyOptOut,
+		Namespace: infra.Namespace,
+		Params: inject.Params{
+			InitImage:         inject.InitImageName(infra.Hub, infra.Tag),
+			ProxyImage:        inject.ProxyImageName(infra.Hub, infra.Tag),
+			Verbosity:         infra.Verbosity,
+			SidecarProxyUID:   inject.DefaultSidecarProxyUID,
+			EnableCoreDump:    true,
+			Version:           "integration-test",
+			Mesh:              mesh,
+			MeshConfigMapName: "istio",
+		},
+	}
+
+	// NOTE: InitializerConfiguration is cluster-scoped and may be
+	// created and used by other tests in the same test cluster.
+	if infra.UseInitializer {
 		if err := deploy("initializer-config.yaml.tmpl", infra.IstioNamespace); err != nil {
```

Comment:
can we serialize inject.Config to a template or post it with k8s API? is that valuable?

Reply:
The sidecar parameters (inject.Config) are available in the template [here](https://github.com/istio/pilot/pull/1104/files/#diff-2d12068fb87af5b21a9140990090b41a). Or did you mean something else?

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: istio-inject
data:
  config: |-
    policy: {{.Policy}}
    namespace: {{.Namespace}}
    params:
      initImage: {{.Params.InitImage}}
      proxyImage: {{.Params.ProxyImage}}
      verbosity: {{.Params.Verbosity}}
      SidecarProxyUID: {{.Params.SidecarProxyUID}}
      Version: {{.Params.Version}}
      EnableCoreDump: {{.Params.EnableCoreDump}}
      MeshConfigMapName: {{.Params.MeshConfigMapName}}
      ImagePullPolicy: {{.Params.ImagePullPolicy}}
      IncludeIPRanges: {{.Params.IncludeIPRanges}}
```